src/dicognito/__main__.py

```diff
@@ -57,16 +65,35 @@ def main(args=None):
 
     args = parser.parse_args(args)
 
+    numeric_level = getattr(logging, args.log_level.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError("Invalid log level: %s" % args.log_level)
+    logging.basicConfig(format="", level=numeric_level)
+
     anonymizer = Anonymizer(id_prefix=args.id_prefix, id_suffix=args.id_suffix, salt=args.salt)
 
     ConvertedStudy = collections.namedtuple("ConvertedStudy", ["AccessionNumber", "PatientID", "PatientName"])
 
+    def get_files_from_source(source):
+        if os.path.isfile(source):
+            yield source
+        elif os.path.isdir(source):
+            for (dirpath, dirnames, filenames) in os.walk(source):
+                for filename in filenames:
+                    yield os.path.join(dirpath, filename)
+        else:
+            for expanded_source in glob.glob(source):
+                for file in get_files_from_source(expanded_source):
+                    yield file
+
     converted_studies = set()
-    for pattern in args.patterns:
-        for file in glob.glob(pattern):
+    for source in args.sources:
+        for file in get_files_from_source(source):
             with pydicom.dcmread(file, force=True) as dataset:
+                if "SOPClassUID" not in dataset:
+                    logging.info("File %s appears not to be DICOM. Skipping.", file)

```

Comment:
There is a `pydicom.misc.is_dicom(file_path)` if that helps here.

Reply:
Yeah, but that's the same check that `dcmread` does when `force` is `False`, which is the way I was thinking of going. I'm a little afraid that we actually care about things that _are_ preambleless but otherwise legal DICOM, and this will require some extra contortions, but for now I'll just go to the other side of the `force`.